src/udiskslinuxnvmecontroller.c

```diff
@@ -174,6 +179,75 @@ update_smart (UDisksLinuxNVMeController *ctrl,
       bd_nvme_smart_log_free (smart_log);
       g_ptr_array_free (a, TRUE);
     }
+  else
+    {
+      /* fallback, smart_log has never been retrieved successfully */
+      const gchar * const *warning = { NULL };
+
+      udisks_nvme_controller_set_smart_critical_warning (UDISKS_NVME_CONTROLLER (ctrl), warning);
+      udisks_nvme_controller_set_smart_power_on_hours (UDISKS_NVME_CONTROLLER (ctrl), 0);
+      udisks_nvme_controller_set_smart_temperature (UDISKS_NVME_CONTROLLER (ctrl), 0);
+    }
+
+  if (selftest_log != NULL)
+    {
+      const gchar *status = "success";
+      gint compl = -1;
+
+      if (selftest_log->current_operation != BD_NVME_SELF_TEST_ACTION_NOT_RUNNING)
+        {
+          compl = 100 - selftest_log->current_operation_completion;
+          status = "inprogress";
+        }
+      else
+      if (selftest_log->entries && *selftest_log->entries)
+        {
+          switch ((*selftest_log->entries)->result)
+            {
+              case BD_NVME_SELF_TEST_RESULT_NO_ERROR:
```

Comment:
We might want to add a new function to libblockdev to get the status strings/descriptions from `BDNVMESelfTestResult`. We already have some "to string" methods in libblockdev so it might make the code simpler.

Reply:
Done, requires https://github.com/storaged-project/libblockdev/pull/755, won't compile in CI at the moment.